lib/queuePopulator/LogReader.js

```diff
@@ -290,14 +290,22 @@ class LogReader {
             timeoutMs: params.timeoutMs,
             logger: this.log.newRequestLogger(),
         };
-
+        const startTime = Date.now();
         async.waterfall([
             next => this._processReadRecords(params, batchState, next),
             next => this._processPrepareEntries(batchState, next),
             next => this._processPublishEntries(batchState, next),
             next => this._processSaveLogOffset(batchState, next),
         ],
             err => {
+                if (this._metricsHandler && this._metricsHandler.processingLag) {
+                    const processingTime = (Date.now() - startTime) / 1000;
+                    const origin = process.env.BACKBEAT_QUEUEPOPULATOR_EXTENSIONS === 'notification' ?
+                    'notification' : 'replication';
+                    this._metricsHandler.processingLag.observe({
```

Comment:
i am wondering if this metric is really specific, and/or if we do not already have something similar/related...

* If it is in LogReader, it should be generic for each QueuePopulator. To make it speciifc to NotificationPopulator, it should be in the NotificationPopulatorExtension...
* Since it is (theorically) possible to have multiple extensions, as is the case with the "usual" QueuePopulator, I am wondering if this "batch" request duration is useful (at QueuePopulator level: e.g. just check how fast we go through the batch?) or needed for bucket notification (what specifically do we want to monitor for bucket notification, e.g. what could go wrong?)

In any case, it seems `this._metricsHandler && this._metricsHandler.processingLag` should be true, even if some tests need to be updated?

Reply:
I decided to remove this metric and to instead watch how much time it takes to get a bucket's config from MongoDB, this was what i originally intended to meter with that metric